test/rules/no-unnecessary-type-assertion/test.ts.fix

```diff
@@ -69,3 +69,13 @@ function func(aOrB: TypeA|TypeB) {
         let z = aOrB;
     }
 }
+
+// Expecting no warning for these assertions as they are not unnecessary.
+
+type Bar = 'bar';
+const data = {
+    x: 'foo' as 'foo',
+    y: 'bar' as Bar,
```

Comment:
I don't get why you need this behavior. A type annotation would also work

```ts
const data = {x: 'foo', y: Bar} = {x: 'foo', y: bar}
```

Same goes for the arrow function below: just add the return type:
```ts
[1, 2, 3, 4, 5].map((x): [number, string] => [x, 'A' + x]);
```


I guess it's useful if the assertion is only a little part of a bigger object where you don't want to write the whole type?